cpp/tests/utilities/column_utilities.cu

```diff
@@ -290,6 +290,20 @@ std::vector<bitmask_type> bitmask_to_host(cudf::column_view const& c)
   }
 }
 
+template <typename T, typename std::enable_if_t<std::is_integral<T>::value>* = nullptr>
+static auto numeric_to_string_precise(T value)
+{
+  return std::to_string(value);
+}
+
+template <typename T, typename std::enable_if_t<std::is_floating_point<T>::value>* = nullptr>
+static auto numeric_to_string_precise(T value)
+{
+  std::ostringstream o;
+  o << std::setprecision(std::numeric_limits<T>::max_digits10) << value;
```

Comment:
This should be `std::setprecision(std::numeric_limits<T>::max_digits10 + 1)`, otherwise certain values cannot be accurately represented. Going to open a separate PR to fix.

Reply:
Which values are problematic here?

Reply:
I was mistaken.

I previously tested with both `digits10` and `max_digits10`, and recalled digits10 needing at least `+1`.

It's also worth noting that the value returned by `numeric_to_string_precise` is still an approximation of the double, but when parsed with std::stod, the resulting double should be bit accurate with the original.

`max_digits10` is sufficient by definition: https://en.cppreference.com/w/cpp/types/numeric_limits/max_digits10